lansa-vmss-windows-autoscale-sql-database/azuredeploy.json

```diff
@@ -676,6 +726,147 @@
+         "properties": {
+            "publicIPAllocationMethod": "Dynamic",
+            "dnsSettings": {
+               "domainNameLabel": "[variables('aglongNamingInfix')]"
+            }
+         }
+      },
+      {
+         "apiVersion": "2017-06-01",
+         "name": "[variables('agName')]",
+         "type": "Microsoft.Network/applicationGateways",
+         "location": "[resourceGroup().location]",
+         "dependsOn": [
+            "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]",
+            "[concat('Microsoft.Network/publicIPAddresses/', variables('agPublicIPAddressName'))]"
+         ],
+         "properties": {
+            "sku": {
+               "name": "[parameters('applicationGatewaySkuName')]",
+               "tier": "Standard",
+               "capacity": "[parameters('applicationGatewayCapacity')]"
+            },
+            "sslCertificates": [
+               {
+                  "name": "appGatewaySslCert",
+                  "properties": {
+                     "data": "[parameters('certificateBase64Encoded')]",
+                     "password": "[parameters('certificatePassword')]"
+                  }
+               }
+            ],
+            "gatewayIPConfigurations": [
+               {
+                  "name": "appGatewayIpConfig",
+                  "properties": {
+                     "subnet": {
+                        "id": "[variables('agSubnetID')]"
+                     }
+                  }
+               }
+            ],
+            "frontendIPConfigurations": [
+               {
+                  "name": "appGatewayFrontendIP",
+                  "properties": {
+                     "PublicIPAddress": {
+                        "id": "[variables('agPublicIPAddressID')]"
+                     }
+                  }
+               }
+            ],
+            "frontendPorts": [
+               {
+                  "name": "appGatewayFrontendPort",
+                  "properties": {
+                     "Port": 443
+                  }
+               }
+            ],
+            "backendAddressPools": [
+               {
+                  "name": "[variables('agBackendAddressPoolName')]"
+               }
+            ],
+            "backendHttpSettingsCollection": [
+               {
+                  "name": "appGatewayBackendHttpSettings",
+                  "properties": {
+                     "Port": 80,
+                     "Protocol": "Http",
+                     "CookieBasedAffinity": "Enabled",
+                     "RequestTimeout": 120
+                  }
+               }
+            ],
+            "httpListeners": [
+               {
+                  "name": "appGatewayHttpListener",
+                  "properties": {
+                     "FrontendIPConfiguration": {
+                        "Id": "[concat(variables('agID'), '/frontendIPConfigurations/appGatewayFrontendIP')]"
```

Comment:
remove concat, use resourceId

Reply:
Please provide information, maybe a link to a documentation resource, which describes how to reference the property of a template resource that is not the id of the template resource using the Resource Id. Obtaining the id is trivial. Any other property is opaque. Most of the comments you have made are for such types of references.